docs/hazmat/primitives/key-derivation-functions.rst

```diff
@@ -600,6 +600,154 @@ Different KDFs are suitable for different tasks such as:
         raises an exception if they do not match.
 
 
+.. currentmodule:: cryptography.hazmat.primitives.kdf.kbkdf
+
+.. class:: Mode
```

Comment:
cc @alex on naming here. Should we leave these `Mode` and `CounterLocation` or should we call them something like `KDFMode`. They're already in the kdf package so I guess I think their names are fine.